Docs: Create and update S2I procedures for the Deploying on OpenShift guide (#45954)
quarkusio/quarkus: docs/src/main/asciidoc/deploying-to-openshift-S2I-howto.adoc

> ////

Reviewer: Should the filename use only lower-case characters?

----
oc import-image {name-image-ubi8-open-jdk-17-short} --from={name-image-ubi8-open-jdk-17} --confirm
----
+
[NOTE]
====
* If you are using the OpenShift image registry and pulling from image streams in the same project, your pod service account should already have the correct permissions.
* If you are pulling images across other {openshift} projects or from secured registries, additional configuration steps might be required.

For more information, see link:https://docs.openshift.com/container-platform/[Red Hat Openshift Container Platform] documentation.
====

. To build the project, create the application, and deploy the {openshift} service, enter the following command:
+
[source,xml,subs="attributes+,+quotes"]
----
oc new-app registry.access.redhat.com/ubi8/openjdk-17~<git_path> --name=<project_name>
----
+
Where:

* <git_path> is the path to the Git repository that hosts your Quarkus project.
For example, `oc new-app registry.access.redhat.com/ubi8/openjdk-17~https://github.com/johndoe/code-with-quarkus.git --name=code-with-quarkus`.
+
If you do not have SSH keys configured for the Git repository, when specifying the Git path, use the HTTPS URL instead of the SSH URL.

> * <project_name> is the name of your application.

Reviewer: ```suggestion
* Replace `<project_name>` with the name of your application.
```

> . To deploy an updated version of the project, push any updates to the Git repository then enter the following command:

Reviewer: ```suggestion
. To deploy an updated version of the project, push changes to the Git repository, and then run:
```

+
[source,xml,subs="attributes+,+quotes"]
----
oc start-build <project_name>
----
+

> . To expose a route to the application, enter the following command:

Reviewer: ```suggestion
. Optional: To expose a route to the application, enter the following command:
```

+
[source,shell,subs="attributes+,+quotes"]
----
oc expose svc <project_name>
----
+

=== Verification

> . To view a list of pods associated with your current OpenShift project, enter the following command:

Reviewer: ```suggestion
. List the pods associated with your current {openshift} project:
```

+
[source,shell,subs="attributes+,+quotes"]
----
oc get pods
----

> . To retrieve the log output for your application's pod, enter the following command where `<pod_name>` is the name of the latest pod prefixed with the name of your application:

Reviewer: ```suggestion
To retrieve the log output for your application's pod, run the following command, replacing `<pod_name>` with the name of the latest pod prefixed by your application name:
```

Reviewer: Similar suggestions apply to the following section.
